src/Dapr.Client/DaprClientGrpc.cs

```diff
@@ -1374,6 +1374,118 @@ private Autogenerated.StateItem ToAutogeneratedStateItem(StateTransactionRequest
 
         #endregion
 
+
+        #region Workflow API
+        /// <inheritdoc/>
+        [Obsolete]
+        public async override Task<WorkflowReference> StartWorkflow(
+            string instanceId,
+            string workflowComponent,
+            string workflowType,
+            Object input,
+            IReadOnlyDictionary<string, string> workflowOptions = default,
+            CancellationToken cancellationToken = default)
+        {
+            ArgumentVerifier.ThrowIfNullOrEmpty(instanceId, nameof(instanceId));
+            ArgumentVerifier.ThrowIfNullOrEmpty(workflowComponent, nameof(workflowComponent));
+            ArgumentVerifier.ThrowIfNullOrEmpty(workflowType, nameof(workflowType));
+            ArgumentVerifier.ThrowIfNull(workflowOptions, nameof(workflowOptions));
+            ArgumentVerifier.ThrowIfNull(input, nameof(input));
+
+            var request = new Autogenerated.StartWorkflowRequest()
+            {
+                InstanceId = instanceId,
+                WorkflowComponent = workflowComponent,
+                WorkflowName = workflowType,
+                Input = (ByteString)input
+            };
+
+            if (workflowOptions?.Count > 0)
+            {
+                foreach (var item in workflowOptions)
```

Comment:
Do we really need this `if` here?

Reply:
I think you need it for the null check.

Reply:
It is here to check against the possibility of workflowOptions being Null.